components/spi_flash/spi_flash_os_func_app.c

```diff
@@ -83,7 +84,11 @@ static IRAM_ATTR esp_err_t spi1_start(void *arg)
      * From 1 to 3, the lock efficiency decreases.
      */
 #if SPI_FLASH_CACHE_NO_DISABLE
-    k_mutex_lock(&s_spi1_flash_mutex, K_FOREVER);
+    if (k_is_pre_kernel()) {
+        s_spi1_key = irq_lock();
```

Comment:
Can you get rid of the lock/unlock and pre kernel calls?

Reply:
no I can't. Using the mutex during the pre-kernel init will hang on a memory exception.